### PR TITLE
Go from the checkout review step to the terms and conditions page

### LINKED ISSUE #434: Click T&C from checkout review goes to home (works from registration page)

### PR DESCRIPTION
**Pull request by mszczepaniak**
Previously when Terms&Conditions was clicked, it has not navigated to the proper page.

Closes #434

### CODE REVIEW

File: projects/storefrontapp-e2e/src/smoke/terms-and-conditions.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    await paymentForm.fillIn();
+    await paymentForm.nextButton.click();
+
+    const reviewForm = checkoutPage.reviewForm;
+    await reviewForm.waitForReady();
+    const termsAndConditionsPage = await checkoutPage.openTermsAndConditions();
+    await termsAndConditionsPage.waitForReady();
+
+    expect(await browser.getCurrentUrl()).toBe(
+      'http://localhost:4200/terms-and-conditions'
```
Comment (Matejk00) [inline projects/storefrontapp-e2e/src/smoke/terms-and-conditions.e2e-spec.ts]: Using constant string with absolute url like this is not good idea :) Please find some dynamic way to check expected result.

File: projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/container/multi-step-checkout.component.html [hunk 2]
```diff
@@ -106,7 +106,11 @@ <h1 class="cx-page__title d-none d-lg-block d-xl-block">
                 />
                 <span class="form-check-label">
                   I confirm that I have read and agree with the
-                  <a href="#">Terms & Conditions</a>
+                  <a
+                    href="/terms-and-conditions"
```
Comment (Matejk00) [inline projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/container/multi-step-checkout.component.html]: I think we can use `routerLink` instead basic href. What you think?

Comment (mszczepaniak) [reply projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/container/multi-step-checkout.component.html]: Sure, good idea, will refactor

File: projects/storefrontapp-e2e/src/smoke/terms-and-conditions.e2e-spec.ts (hunk 1 above)
Comment (mszczepaniak) [reply projects/storefrontapp-e2e/src/smoke/terms-and-conditions.e2e-spec.ts]: Well, this expectation is i.e. in the register E2E spec, so I have assumed that it's ok. I will try the dynamic way

Comment (Matejk00) [reply projects/storefrontapp-e2e/src/smoke/terms-and-conditions.e2e-spec.ts]: It shouldn't be like this. For example if you try to run e2e test on different port f.e. `ng e2e --port 4201` than such tests always be failing.

Comment (mszczepaniak) [reply projects/storefrontapp-e2e/src/smoke/terms-and-conditions.e2e-spec.ts]: Refactored not to evaluate against port, used toContain

**Pull request merged**